### PR TITLE
Vim: bump version & improve recipe

### LINKED ISSUE #4894: Vim: It's not possible to 'exit' terminal
Steps:

1. Open vim
2. Type `:term`
3. Type `exit` on terminal window

It should exit, but it just echoes the command, and hangs.

### PR DESCRIPTION
**Pull request by bitigchi**
This PR updates Vim to 8.2.0671. This version includes support for installing Vim without dependence on a HaikuPorts recipe via passing commands about assets handling to the Vim Makefile. Several build warnings have been tackled, and some duplicate GUI code has been eliminated. Documentation has been improved as well.

In addition to this, I believe it's not necessary to compile Vim twice in this recipe. Separating GUI compile from the CLI compile prevents Vim from using the `+clipboard` function, therefore it is not possible to use the system clipboard on the console version. By compiling Vim with GUI support only, this feature is available on both ends.

I have been testing this for the last couple of days, and haven't encountered any issues.

Fixes #4894, #4895, #4907

### CODE REVIEW

**Comment by pulkomandy** [comment]
On Thu, Apr 30, 2020 at 02:24:59PM -0700, Emir Sarı wrote:
> In addition to this, I believe it&#39;s not necessary to compile Vim twice in this recipe. Separating GUI compile from the CLI compile prevents Vim from using the `+clipboard` function, therefore it is not possible to use the system clipboard on the console version. By compiling Vim with GUI support only, this feature is available on both ends.

This will make it impossible to run Vim from users without an app_server
connection (for example if you created a new user account and let it
connect to your machine using ssh). Also it shows an entry in DeskBar
(at least we should set the B_BACKGROUND_APP flag on the app when
running in console, I guess).

**Comment by bitigchi** [comment]
Are there any other disadvantages to this except the ssh issue?

**Comment by theclue** [comment]
I use vi mainly from ssh so pls don't ;)

**Comment by bitigchi** [comment]
> I use vi mainly from ssh so pls don't ;)

Who needs vi when you have the ed(1) standard text editor? :P

File: app-editors/vim/vim-8.2.0742.recipe [hunk 1]
```diff
@@ -101,18 +101,17 @@ BUILD()
 # CLI -------------------------------------------
 
 	# Clean objects from possible previous run to avoid mixup of GUI/CLI objects
-	rm objects/*.o || true
+        rm objects/*.o || true
```
Comment (diversys) [inline app-editors/vim/vim-8.2.0742.recipe:104]: Please use tabs.

Comment (bitigchi) [reply app-editors/vim/vim-8.2.0742.recipe]: Fixed.

**Comment by diversys** [comment]
app-editors/vim/.swp shouldn't be there.

**Comment by bitigchi** [comment]
Fixed that, slipped through my eyes.

**Pull request merged**